ledger/accountdb.go

```diff
@@ -216,6 +394,109 @@ func (qs *accountsDbQueries) lookup(addr basics.Address) (data basics.AccountDat
 	return
 }
 
+func (qs *accountsDbQueries) storeCatchpoint(ctx context.Context, round basics.Round, fileName string, catchpoint string, fileSize int64) (err error) {
+	err = db.Retry(func() (err error) {
+		_, err = qs.deleteStoredCatchpoint.ExecContext(ctx, round)
+
+		if err != nil || (fileName == "" && catchpoint == "" && fileSize == 0) {
```

Comment:
why ? the cpu time it would take to do that isn't substantial enough, and we need to delete the old stored catchpoint anyway.
I think that a better thing to do here is to figure out if we want to perform just delete vs update/insert statement. But I can't see how we get any practical performance gains here.